README.md

```diff
@@ -12,7 +109,44 @@ py -m pip install --upgrade pip
 py -m pip install -r requirements.txt
 ```
 
-`pip freeze > requirements.txt`
 
-Using `requirements.txt` enter following shell commands one by one:
+# How to create requirements.txt
+
+When you push your code to GitHub, you want anyone to be able to download, run it and continue working on it. This means they will have to be able to successfully install all dependancies for your code using `requirements.txt`.
+
+To create the file make sure the correct enviroment is active and use the following command:
+
+```
+pip freeze > requirements.txt
+```
+This will create `requirements.txt` in the root directory with all packages and their dependancies used in your project.
+
+# .gitignore
+
+Before pushing your code to GitHub, sometimes there are files/folders you do not want/need to update.
+
+This includes things like:
+* the enviroment itslef - because it's very large and GitHub doesn't allow updating anything more than 100mb
+* `__pycache__` - cache generated by Pyhon
+* any sensitive files containing credentials/passwords
+* trained weights of models - especially deep learning because those can be in GB sizes
+
+To exclude these from being pushed to GitHub, in the root directory create `.gitignore` file.
+
+Inside create a new line withe whatever you want to exclude from being pushed to GitHub.
+
+Example:
+
+```
+.env/
+__pycache__/
+credentials1.py
+models/credentials2.py
+```
+This will remove all folders and their content with names `.env` and `__pycache__`. 
+
+It will remove `credentials1.py` file in the root directory and `credentials2` in the `models` directory - but will keep everything else in there.
+
+
+Reviewer: please remove this line of code before merging the pull request.
```

Comment:
Is this the line that needs to be removed?

Reply:
yes

Reply:
ok